bigtable/client/metadata_update_policy_test.cc

```diff
@@ -13,13 +13,278 @@
+  }
+
+ private:
+  BigtableImpl bigtable_service_;
+  TableAdminImpl admin_service_;
+  grpc::ServerBuilder builder_;
+  std::unique_ptr<grpc::Server> server_;
+  std::string address_;
+};
+
+/// @test A test for setting metadata when table is not known.
+TEST(MetadataUpdatePolicy, RunWithEmbeddedServer) {
+  std::unique_ptr<DefaultEmbeddedServer> server =
+      bigtable::internal::make_unique<DefaultEmbeddedServer>();
+  std::string address = server->address();
+  EXPECT_FALSE(server->address().empty());
+  bigtable::ClientOptions options;
+  options.set_admin_endpoint(address);
+  options.set_data_endpoint(address);
+  options.SetCredentials(grpc::InsecureChannelCredentials());
+  std::thread wait_thread([&server]() { server->Wait(); });
+  EXPECT_TRUE(wait_thread.joinable());
+
+  // Create the table against embedded server.
+  bigtable::TableAdmin admin(
+      bigtable::CreateDefaultAdminClient("fake-project", options),
+      "fake-instance");
+  grpc::string_ref expected_x_goog_request_param =
+      "parent=projects/fake-project/instances/fake-instance";
+
+  auto gc = bigtable::GcRule::MaxNumVersions(42);
+  admin.CreateTable("fake-table-01", bigtable::TableConfig({{"fam", gc}}, {}));
+
+  // Get the metadata from embedded server
+  auto client_metadata = server->admin_service_client_metadata();
+  std::pair<std::multimap<grpc::string_ref, grpc::string_ref>::iterator,
+            std::multimap<grpc::string_ref, grpc::string_ref>::iterator>
+      result;
+  result = client_metadata.equal_range("x-goog-request-params");
+  // first check to see if client_metadata has only one occurance of parameter.
+  EXPECT_EQ(1, std::distance(result.first, result.second));
+
+  for (std::multimap<grpc::string_ref, grpc::string_ref>::iterator it =
+           result.first;
+       it != result.second; ++it) {
+    grpc::string_ref actual_x_goog_request_param = it->second;
+    EXPECT_EQ(expected_x_goog_request_param, actual_x_goog_request_param);
+  }
+  EXPECT_TRUE(wait_thread.joinable());
+  server->Shutdown();
+  wait_thread.join();
+}
+/// @test A test for setting metadata when table_id is known during the call.
+TEST(MetadataUpdatePolicy, RunWithEmbeddedServerLazyMetadata) {
+  std::unique_ptr<DefaultEmbeddedServer> server =
+      bigtable::internal::make_unique<DefaultEmbeddedServer>();
+  std::string address = server->address();
+  EXPECT_FALSE(server->address().empty());
+  bigtable::ClientOptions options;
+  options.set_admin_endpoint(address);
+  options.set_data_endpoint(address);
+  options.SetCredentials(grpc::InsecureChannelCredentials());
+  std::thread wait_thread([&server]() { server->Wait(); });
+  EXPECT_TRUE(wait_thread.joinable());
+
+  // Create the table against embedded server.
+  bigtable::TableAdmin admin(
+      bigtable::CreateDefaultAdminClient("fake-project", options),
+      "fake-instance");
+
+  admin.GetTable("fake-table-01");
+
+  grpc::string_ref expected_x_goog_request_param =
```

Comment:
Consider just "expected" as a variable name, here and in similar places.

Reply:
fixed